src/com/sap/piper/ConfigurationMerger.groovy

```diff
@@ -6,7 +6,7 @@ import com.cloudbees.groovy.cps.NonCPS
 class ConfigurationMerger {
     @NonCPS
     static Map merge(Map configs, Set configKeys, Map defaults) {
-        Map filteredConfig = configKeys?configs.subMap(configKeys):configs
+        Map filteredConfig = configKeys==null?configs:configKeys.size()==0?[:]:configs.subMap(configKeys)
```

Comment:
Please format the expression with spaces so that it makes sense. Hard to read as is.